File: ru.kmorozov.gbd.core.config/src/main/java/ru/kmorozov/gbd/core/logic/model/book/base/AbstractPage.java
package ru.kmorozov.gbd.core.logic.model.book.base;

import java.util.concurrent.atomic.AtomicBoolean;

/**
 * Created by km on 18.11.2016.
 */
public abstract class AbstractPage implements IPage {

    private final AtomicBoolean sigChecked = new AtomicBoolean(false);
    private final AtomicBoolean dataProcessed = new AtomicBoolean(false);
    private final AtomicBoolean fileExists = new AtomicBoolean(false);
    private final AtomicBoolean loadingStarted = new AtomicBoolean(false);

    @Override
    public boolean isDataProcessed() {
        return this.dataProcessed.get();
    }

    @Override
    public boolean isFileExists() {
        return this.fileExists.get();
    }

    @Override
    public boolean isSigChecked() {
        return this.sigChecked.get();
    }

    @Override
    public boolean isLoadingStarted() {
        return this.loadingStarted.get();
    }

    public boolean isProcessed() {
        return this.isDataProcessed() || this.sigChecked.get() || this.isFileExists();
    }

    public boolean isNotProcessed() {
        return !this.isProcessed();
    }

    public abstract String getImgUrl();

    public void setSigChecked(final boolean value) {
        this.sigChecked.set(value);
    }

    public void setDataProcessed(final boolean value) {
        this.dataProcessed.set(value);
    }

    public void setFileExists(final boolean value) {
        this.fileExists.set(value);
    }

    public void setLoadingStarted(final boolean value) {
        this.loadingStarted.set(value);
    }
}

File: ru.kmorozov.gbd.core.config/src/main/java/ru/kmorozov/gbd/core/logic/model/book/base/AbstractPage.kt
package ru.kmorozov.gbd.core.logic.model.book.base

import ru.kmorozov.gbd.core.config.IStoredItem
import java.util.concurrent.atomic.AtomicBoolean

/**
 * Created by km on 18.11.2016.
 */
abstract class AbstractPage : IPage {

    private val dataProcessed = AtomicBoolean(false)
    private val fileExists = AtomicBoolean(false)
    public val loadingStarted = AtomicBoolean(false)
    private val scanned = AtomicBoolean(false)

    override var isDataProcessed: Boolean
        get() = dataProcessed.get()
        set(value) = dataProcessed.set(value)

    override var isFileExists: Boolean
        get() = fileExists.get()
        set(value) = fileExists.set(value)

    override var isLoadingStarted: Boolean
        get() = loadingStarted.get()
        set(value) = loadingStarted.set(value)

    override var isScanned: Boolean
        get() = scanned.get()
        set(value) = scanned.set(value)

    val isProcessed: Boolean
        get() = isDataProcessed || isFileExists

    val isNotProcessed: Boolean
        get() = !isProcessed

    abstract val imgUrl: String

    val isGapPage: Boolean
        get() = false

    override lateinit var storedItem: IStoredItem
}
